tv_apps/smart-system/js/app_install_manager.js

```diff
@@ -292,6 +322,72 @@ var AppInstallManager = {
     app.onprogress = this.handleProgress;
   },
 
+  previewApp: function ai_previewApp(app) {
+    if (!this.isMarketPlaceAppActive()) {
+      return;
+    }
+
+    var marketplace = AppWindowManager.getActiveApp();
+    var manifestURL = app.manifestURL;
+    var appURL = app.origin +
+      (app.manifest.launch_path === '/' ? '/' : app.manifest.launch_path);
+
+    marketplace.childWindowFactory.createPreviewWindow({
+      detail: {
+        url: appURL,
+        origin: app.origin,
+        manifestURL: manifestURL
+      }
+    });
+
+    marketplace.element.addEventListener('_closed', (evt) => {
+      this.clearPreviewApps();
+    });
+
+    var previewAppWindow = marketplace.getTopMostWindow();
```

Comment:
Is there any possibility that the `previewAppWindow` isn't created correctly? Do we need to check the returned value of `createPreviewWindow`?